src/shared/install.c

```diff
@@ -2873,7 +2873,10 @@ static int do_unit_file_disable(
                 r = install_info_add(&ctx, *name, NULL, lp->root_dir, /* auxiliary= */ false, &info);
                 if (r >= 0)
                         r = install_info_traverse(&ctx, lp, info, SEARCH_LOAD|SEARCH_FOLLOW_CONFIG_SYMLINKS, NULL);
-
+                if (r == -ENOENT)
+                        /* In case there's no unit, we still want to remove any leftover symlink, even if
+                         * the unit might have been removed already. */
```

Comment:
I think we should still report this enoent via the install changes mechanism, hence I think this should be changed like above, i.e. change the below to:

```c
if (r < 0)
        r = install_changes(…, r, …);
        if (r != -ENOENT)
                return r;
```